databricks/koalas/groupby.py

```diff
@@ -928,6 +928,15 @@ def apply(self, func):
                 pser_or_pdf = pdf.groupby(input_groupnames)[name].apply(func)
             else:
                 pser_or_pdf = pdf.groupby(input_groupnames).apply(func)
+                keys = set(input_groupnames)
+                should_drop_columns = (
+                    isinstance(pser_or_pdf, pd.DataFrame)
+                    and keys.issubset(set(pser_or_pdf.index.names))
+                    and keys.issubset(set(pser_or_pdf.columns))
+                )
+                if should_drop_columns:
+                    pser_or_pdf = pser_or_pdf.drop(input_groupnames, axis=1)
```

Comment:
Can you show before/after output in the PR description? From a cursory look, seems like it will just drop the duplicated column which isn't consistent with pandas' results.

Reply:
I agree with @HyukjinKwon 's opinion since I think we should match the Koalas behavior to pandas' exactly same without any additional post-processing.

Reply:
I updated the PR description, looking forward to your comments